ncm-systemd: alias sys-device-pci  doesn't match expected pattern (EL9)

I'm trying to install an EL9 server (Alma 9.3) and when running `ncm-systemd`, I get the following errors: 

```
2024/04/02-11:37:57 [VERB] [INFO] running component: systemd
2024/04/02-11:37:57 [VERB] ---------------------------------------------------------
2024/04/02-11:37:59 [VERB] [ERROR] Found alias "sys-devices-pci0000:00-0000:00:1a.0-usb1-1\\x2d1-1\\x2d1.6-1\\x2d1.6.3.device" for unit sys-devices-pci0000:00-0000:00:1a.0-usb1-1\x2d1-1\x2d1.6-1\x2d1.6.3.device that doesn't match expected pattern '^(.*)\.(automount|device|mount|path|scope|service|slice|snapshot|socket|swap|target|timer)$'. Skipping.
2024/04/02-11:37:59 [VERB] [ERROR] get_unit_show: no alias for unit sys-devices-pci0000:00-0000:00:1a.0-usb1-1\x2d1-1\x2d1.6-1\x2d1.6.3.device defined. (Forgot to update cache?)
2024/04/02-11:38:00 [VERB] [ERROR] Found alias "sys-devices-pci0000:00-0000:00:1a.0-usb1-1\\x2d1-1\\x2d1.6-1\\x2d1.6.3.device" for unit sys-devices-pci0000:00-0000:00:1a.0-usb1-1\x2d1-1\x2d1.6-1\x2d1.6.3.device that doesn't match expected pattern '^(.*)\.(automount|device|mount|path|scope|service|slice|snapshot|socket|swap|target|timer)$'. Skipping.
2024/04/02-11:38:00 [VERB] [ERROR] get_unit_show: no alias for unit sys-devices-pci0000:00-0000:00:1a.0-usb1-1\x2d1-1\x2d1.6-1\x2d1.6.3.device defined. (Forgot to update cache?)
2024/04/02-11:38:01 [VERB] [INFO] Configure on component systemd executed, 4 errors, 0 warnings
```

When looking at the details in `component/systemd.log`, I find:

```
2024/04/02-11:15:20 [VERB] Getting output of command: /usr/bin/systemctl --all --no-pager --no-legend --full list-units
2024/04/02-11:15:20 [VERB] Getting output of command: /usr/bin/systemctl --all --no-pager --no-legend --full list-unit-files
2024/04/02-11:15:20 [VERB] Getting output of command: /usr/bin/systemctl --no-pager --all show -- sys-devices-pci0000:00-0000:00:1a.0-usb1-1\x2d1-1\x2d1.6-1\x2d1.6.3.device
2024/04/02-11:15:20 [ERROR] Found alias "sys-devices-pci0000:00-0000:00:1a.0-usb1-1\\x2d1-1\\x2d1.6-1\\x2d1.6.3.device" for unit sys-devices-pci0000:00-0000:00:1a.0-usb1-1\x2d1-1\x2d1.6-1\x2d1.6.3.device that doesn't match expected pattern '^(.*)\.(automount|device|mount|path|scope|service|slice|snapshot|socket|swap|target|timer)$'. Skipping.
2024/04/02-11:15:20 [VERB] make_cache_alias completed with 328 cached units 37 alias units
2024/04/02-11:15:20 [ERROR] get_unit_show: no alias for unit sys-devices-pci0000:00-0000:00:1a.0-usb1-1\x2d1-1\x2d1.6-1\x2d1.6.3.device defined. (Forgot to update cache?)
2024/04/02-11:15:20 [VERB] Undefined UnitFileState for unit sys-devices-pci0000:00-0000:00:1a.0-usb1-1\x2d1-1\x2d1.6-1\x2d1.6.3.device
```

Environement used:
- AlmaLinux 9.3
- Quattor 23.6.0 (`ncm-systemd-23.6.0-1.noarch`)

for reference, we also get a similar error while configuring qemu-guest-agent in systemd:
```
2024/04/25-13:12:59 [VERB] Getting output of command: /usr/bin/systemctl --all --no-pager --no-legend --full list-units
2024/04/25-13:12:59 [VERB] Getting output of command: /usr/bin/systemctl --all --no-pager --no-legend --full list-unit-files
2024/04/25-13:13:00 [VERB] Getting output of command: /usr/bin/systemctl --no-pager --all show -- sys-devices-pci0000:00-0000:00:06.0-virtio3-virtio\x2dports-vport3p1.device
2024/04/25-13:13:00 [ERROR] Found alias "sys-devices-pci0000:00-0000:00:06.0-virtio3-virtio\\x2dports-vport3p1.device" for unit sys-devices-pci0000:00-0000:00:06.0-virtio3-virtio\x2dports-vport3p1.device that doesn't match expected pattern '^(.*)\.(automount|device|mount|path|scope|service|slice|snapshot|socket|swap|target|timer)$'. Skipping.
2024/04/25-13:13:00 [VERB] Getting output of command: /usr/bin/systemctl --no-pager --all show -- dev-virtio\x2dports-org.qemu.guest_agent.0.device
2024/04/25-13:13:00 [ERROR] Found alias "dev-virtio\\x2dports-org.qemu.guest_agent.0.device" for unit dev-virtio\x2dports-org.qemu.guest_agent.0.device that doesn't match expected pattern '^(.*)\.(automount|device|mount|path|scope|service|slice|snapshot|socket|swap|target|timer)$'. Skipping.
2024/04/25-13:13:00 [VERB] make_cache_alias completed with 358 cached units 46 alias units
2024/04/25-13:13:00 [ERROR] get_unit_show: no alias for unit dev-virtio\x2dports-org.qemu.guest_agent.0.device defined. (Forgot to update cache?)
2024/04/25-13:13:00 [VERB] Undefined UnitFileState for unit dev-virtio\x2dports-org.qemu.guest_agent.0.device
2024/04/25-13:13:00 [ERROR] get_unit_show: no alias for unit sys-devices-pci0000:00-0000:00:06.0-virtio3-virtio\x2dports-vport3p1.device defined. (Forgot to update cache?)
2024/04/25-13:13:00 [VERB] Undefined UnitFileState for unit sys-devices-pci0000:00-0000:00:06.0-virtio3-virtio\x2dports-vport3p1.device
```

we have also hit this issue now. seeing the following

`2024/06/03-12:26:11 [ERROR] Found alias "dev-disk-by\\x2duuid-be3848db\\x2d217a\\x2d4574\\x2d80e0\\x2d67813f0d4407.device" for unit dev-disk-by\x2duuid-be3848db\x2d217a\x2d4574\x2d80e0\x2d67813f0d4407.device that doesn't match expected pattern '^(.*)\.(automount|device|mount|path|scope|service|slice|snapshot|socket|swap|target|timer)$'. Skipping.`

Are we close to any fix yet?

--- a/ncm-systemd/src/main/perl/Systemd/Systemctl.pm
+++ b/ncm-systemd/src/main/perl/Systemd/Systemctl.pm
@@ -148,8 +148,19 @@ sub systemctl_show
     while($output =~ m/^([^=\s]+)\s*=(.*)?$/mg) {
         my ($k,$v) = ($1,"$2");
         if (grep {$_ eq $k} @PROPERTIES_ARRAY) {
-            my @values = split(/\s+/, $v);
-            $res->{$k} = \@values;
+            my $values = [];
+            foreach my $value (split(/\s+/, $v)) {
+                if ($value =~ m/^"(.*)"$/) {
+                    $value = $1;
+                    if ($value =~ m/\\{2}/) {
+                        $value =~ s/\\{2}/\\/g;
+                    } else {
+                        $logger->error("Found double quoted value '$value' but no expected quoting pattern from '$k = $v'")
+                    }
+                }
+                push(@$values, $value);
+            };
+            $res->{$k} = $values;
         } else {
             $res->{$k} = $v;
         }

